src/store/product/reducer.js

```diff
@@ -0,0 +1,26 @@
+import actions from './actionTypes';
+
+const product = {};
+
+const initialState = product ? { isLoading: false, product } : {};
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case actions.PRODUCT_REQUEST:
+      return {
+        isLoading: true,
+        product: action.product,
```

Comment:
you dont need to save the product here since at this point it's not actually saved in the backend